#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "../protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Roughcast.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.5559520593300852 -0.47181208610698055 -0.6843322753823736 4.031822089419813
  position 1.1859528645318833 -1.5773610448559956 5.50042624930142
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2.5 2.5
  floorAppearance PBRAppearance {
    baseColor 0.9 0.7 0.6
    roughness 1
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
}
PointLight {
  attenuation 0 0 1
  intensity 0.98
  location 0.6 0.34 0.4
}
Wall {
  translation -1 1.125 0
  name "1"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.5 -1.125 0
  name "1(1)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.25 -0.875 0
  name "1(3)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.75 -0.875 0
  name "1(4)"
  size 0.01 0.25 0.1
}
Wall {
  translation 1 -0.875 0
  name "1(5)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.25 -0.875 0
  name "1(6)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.25 0.875 0
  name "1(7)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.25 0.125 0
  name "1(8)"
  size 0.01 0.25 0.1
}
Wall {
  translation 1.125 0.75 0
  rotation 0 0 1 1.5708003061004252
  name "1(9)"
  size 0.01 0.25 0.1
}
Wall {
  translation 1.125 0.5 0
  rotation 0 0 1 1.5708003061004252
  name "1(10)"
  size 0.01 0.25 0.1
}
Wall {
  translation 1.125 -0.75 0
  rotation 0 0 1 1.5708003061004252
  name "1(11)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.875 1 0
  rotation 0 0 1 1.5708003061004252
  name "1(12)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.375 0.25 0
  rotation 0 0 1 1.5708003061004252
  name "1(13)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.125 0.75 0
  rotation 0 0 1 1.5708003061004252
  name "1(14)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.625 -0.25 0
  rotation 0 0 1 1.5708003061004252
  name "1(15)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.875 0.25 0
  rotation 0 0 1 1.5708003061004252
  name "1(16)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.875 0.75 0
  rotation 0 0 1 1.5708003061004252
  name "1(17)"
  size 0.01 0.25 0.1
}
Wall {
  translation -1.125 -1 0
  rotation 0 0 1 1.5708003061004252
  name "1(18)"
  size 0.01 0.25 0.1
}
Wall {
  translation -1.125 -0.5 0
  rotation 0 0 1 1.5708003061004252
  name "1(19)"
  size 0.01 0.25 0.1
}
Wall {
  translation 0.5 -1.125 0
  name "1(20)"
  size 0.01 0.25 0.1
}
Wall {
  translation -0.75 0.75 0
  name "2"
  size 0.01 0.5 0.1
}
Wall {
  translation 0 0.5 0
  name "2(1)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0 -1 0
  name "2(2)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0.5 1 0
  name "2(3)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0.5 -0.25 0
  name "2(4)"
  size 0.01 0.5 0.1
}
Wall {
  translation 1 0 0
  name "2(5)"
  size 0.01 0.5 0.1
}
Wall {
  translation 1 -0.25 0
  rotation 0 0 1 1.5707996938995747
  name "2(6)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0.75 -0.5 0
  rotation 0 0 1 1.5707996938995747
  name "2(7)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0.5 -0.75 0
  rotation 0 0 1 1.5707996938995747
  name "2(8)"
  size 0.01 0.5 0.1
}
Wall {
  translation 0.25 -0.25 0
  rotation 0 0 1 1.5707996938995747
  name "2(9)"
  size 0.01 0.5 0.1
}
Wall {
  rotation 0 0 1 1.5707996938995747
  name "2(10)"
  size 0.01 0.5 0.1
}
Wall {
  translation -0.25 -0.125 0
  name "3"
  size 0.01 0.75 0.1
}
Wall {
  translation -0.125 -0.5 0
  rotation 0 0 1 -1.5708003061004252
  name "3(1)"
  size 0.01 0.75 0.1
}
Wall {
  translation -0.75 -0.5 0
  name "4"
  size 0.01 1 0.1
}
Wall {
  translation -0.5 0.25 0
  name "4(1)"
  size 0.01 1 0.1
}
Wall {
  translation 0.25 -0.75 0
  name "4(2)"
  size 0.01 1 0.1
}
Wall {
  translation 0.75 0.5 0
  name "4(3)"
  size 0.01 1 0.1
}
Wall {
  translation -0.5 -0.75 0
  rotation 0 0 1 1.5707996938995747
  name "4(3)"
  size 0.01 1 0.1
}
Wall {
  translation -0.25 1 0
  rotation 0 0 1 1.5708003061004252
  name "4(4)"
  size 0.01 1 0.1
}
Wall {
  translation -1 0.125 0
  name "5"
  size 0.01 1.25 0.1
}
Wall {
  translation 0.125 0.5 0
  rotation 0 0 1 1.5707996938995747
  name "5(1)"
  size 0.01 1.25 0.1
}
E-puck {
  translation 0.341765 0.114765 -6.3962e-05
  rotation -2.64209e-10 3.52199e-09 -1 -0.15203530717958635
  controller "my_controller_wallfollowing_new"
}
Wall {
  translation -1.125 -0.49 0
  rotation 0 0 1 1.5708003061004252
  name "green"
  size 0.01 0.1 0.1
  appearance Roughcast {
    colorOverride 0 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.95 -0.24 0
  rotation 0 0 1 1.5707996938995747
  name "brown"
  size 0.01 0.1 0.1
  appearance Roughcast {
    colorOverride 0.647 0.412 0.118
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.05 -1.246 0
  rotation 0 0 1 1.5707996938995747
  name "yellow"
  size 0.01 0.1 0.1
  appearance Roughcast {
    colorOverride 1 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.69 0.49 0
  rotation 0 0 1 1.5707996938995747
  name "red"
  size 0.01 0.1 0.1
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.56 -0.49 0
  rotation 0 0 1 1.5707996938995747
  name "pink"
  size 0.01 0.1 0.1
  appearance Roughcast {
    colorOverride 1 0 1
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
